modules/internal/DefaultRectangular.chpl

```diff
@@ -2243,20 +2265,29 @@ module DefaultRectangular {
   TODO: Pull simple runtime implementation up into module code
   */
   private proc complexTransfer(A, aView, B, bView) {
-    if (A.data.locale.id != here.id &&
-        B.data.locale.id != here.id) {
+    use ChplConfig;
+
+    const Alocid = A.data.locale.id;
+    const Asublocid = if CHPL_LOCALE_MODEL != "gpu" then c_sublocid_any else
+                          chpl_sublocFromLocaleID(A.data.locale.chpl_localeid());
+    const Blocid = B.data.locale.id;
+    const Bsublocid = if CHPL_LOCALE_MODEL != "gpu" then c_sublocid_any else
+                          chpl_sublocFromLocaleID(B.data.locale.chpl_localeid());
```

Comment:
can/should this logic to determine the sublocid be a helper function?

Reply:
For posterity: I did so, but for some reason this increased comm counts, so I rolled back that change.